packages/webapp/src/components/Task/AddSoilAmendmentProducts/QuantityApplicationRate/useQuantityApplicationRate.ts

```diff
@@ -63,8 +63,12 @@ export const useQuantityApplicationRate = ({
     APPLICATION_RATE_VOLUME_UNIT,
   ]);
 
+  useEffect(() => {
+    updateTotalArea(100);
+  }, []);
```

Comment:
Sorry @SayakaOno, it did end up being necessary! Even (a bit to my surprise) the Unit setting logic in the called function below.

Passing to `defaultValues` caused a persistence bug when the selected locations changed. I did rename the function as it is no longer just a handler.

Reply:
Could you provide the steps to reproduce the bug?

Reply:
> Could you provide the steps to reproduce the bug?

Create a soil_amendment task on a farm with multiple locations > go back and change the selected areas (add or remove one) > preview string value will update but the value in the `TOTAL_AREA_AMENDED` `<Unit />` will not update

Reply:
Ohhhh 😳 I hadn't considered those scenarios at all, thank you!
And it makes sense that we need the useEffect. (although I'm not sure about the unit not being set correctly)
I think the argument should be dynamic! (It always becomes 100 regardless of the previous %)

Reply:
> I think the argument should be dynamic! (It always becomes 100 regardless of the previous %)

You mean we could store the previous percent value? I think here we would then have to remove the defaultValue prop on the `<NumberInput /`> 🤔 It could be done with the field array default value worked for the first product as well!

> (although I'm not sure about the unit not being set correctly)

It's weird, right? I originally wrote the useEffect as simply

```ts
  useEffect(() => {
    setValue(TOTAL_AREA_AMENDED, total_area);
  }, []);
  ```
  
but you can see (e.g. in the Imperial Storybook story, or in app depending on your area settings) that it is not picking the right unit for the value!

Reply:
I think I wasn't clear about the first point, so let me share the recording.

https://github.com/user-attachments/assets/8e57ffed-78a6-455b-be54-22c8b5cb4009

Reply:
Ah thank you! I knew it was resetting to 100 but I didn't realize it diverged from the string.